src/System.Net.Http/src/System/Net/Http/OSX/CurlHandler.SslProvider.cs

```diff
@@ -58,24 +71,23 @@ internal static void SetSslOptions(EasyRequest easy, ClientCertificateOption cli
                     // user's keychain and setting the SSL policy trust for it to "Always Trust".
                     // Similarly, the "block this" could be attained by setting the SSL policy for a cert in the
                     // keychain to "Never Trust".
-                    throw new PlatformNotSupportedException(
-                        SR.Format(
-                            SR.net_http_libcurl_callback_notsupported,
-                            CurlVersionDescription,
-                            CurlSslVersionDescription));
-                }
-
-                // Revocation checking is always on for darwinssl (SecureTransport).
-                // If any other backend is used and revocation is requested, we can't guarantee
-                // that assertion.
-                if (easy._handler.CheckCertificateRevocationList &&
-                    !CurlSslVersionDescription.Equals("SecureTransport"))
-                {
-                    throw new PlatformNotSupportedException(
-                    SR.Format(
-                        SR.net_http_libcurl_revocation_notsupported,
-                        CurlVersionDescription,
-                        CurlSslVersionDescription));
+                    //
+                    // However, one case we can support is when we know all certificates will pass validation.
+                    // We can detect a key case of that: whether DangerousAcceptAnyServerCertificateValidator was used.
+                    if (easy.ServerCertificateValidationCallbackAcceptsAll)
+                    {
+                        EventSourceTrace("Disabling peer and host verification per {0}", nameof(HttpClientHandler.DangerousAcceptAnyServerCertificateValidator), easy: easy);
+                        easy.SetCurlOption(Interop.Http.CURLoption.CURLOPT_SSL_VERIFYPEER, 0); // don't verify the peer
+                        easy.SetCurlOption(Interop.Http.CURLoption.CURLOPT_SSL_VERIFYHOST, 0); // don't verify the peer cert's hostname
```

Comment:
I tried removing the CURLOPT_SSL_VERIFYHOST=0 setting.  With an NSS backend, VERIFYPEER=0 automatically implies VERIFYHOST=0 (and there's no way to override that), so it didn't affect the CentOS/Fedora/RHEL VMs with have in CI, as they have an NSS backend.  With a GnuTLS backend on Ubuntu 16.10, and without VERIFYHOST=0, and connecting to a site with an incorrect host name, we get:
```
  [Microsoft-System-Net-Http-12] (65, 64, CurlDebugFunction, CURLINFO_TEXT: subjectAltName does not match wrong.host.badssl.com).
  [Microsoft-System-Net-Http-12] (65, 64, CurlDebugFunction, CURLINFO_TEXT: SSL: no alternative certificate subject name matches target host name 'wrong.host.badssl.com').
```
which then fails the request with:
```
SSL peer certificate or SSH remote key was not OK
```
Similar error for all of our localhost ssl tests.  So I at least need to leave the VERIFYHOST=0 part in for the Linux CurlHandler.SslProvider.cs.  However, CI passed on macOS without setting VERIFYHOST, so maybe we could get away without it?  But I'm not sure why this didn't fail.  With VERIFYHOST at its default value, libcurl should be calling Apple's SSLSetPeerDomainName function:
https://github.com/curl/curl/blob/master/lib/vtls/darwinssl.c#L1536
and that should cause the connection to fail if the hostname doesn't match.